src/auth/services/auth.service.ts

```diff
@@ -60,6 +64,14 @@ export class AuthService {
   ): Promise<RegisterOutput> {
     this.logger.log(ctx, `${this.register.name} was called`);
 
+    // check for existing records
+    if (await this.userService.findByUsername(ctx, input.username)) {
```

Comment:
BadRequestException is nice here. Just throw it for both cases with one message 'User already exists', rather than pin pointing is it username or email. Thanks.

Reply:
Done.